cmd/nerdctl/builder_build_test.go

```diff
@@ -449,3 +449,45 @@ CMD ["cat", "/source-date-epoch"]
 	base.Cmd("build", "-t", imageName, "--build-arg", "SOURCE_DATE_EPOCH="+sourceDateEpochArgStr, buildCtx).AssertOK()
 	base.Cmd("run", "--rm", imageName).AssertOutExactly(sourceDateEpochArgStr + "\n")
 }
+
+func TestBuildNetwork(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").AssertOK()
+	base.Cmd("image", "prune", "--force", "--all").AssertOK()
+
+	dockerfile := fmt.Sprintf(`FROM %s
+RUN apk add --no-cache curl
+RUN curl -I http://google.com
+	`, testutil.CommonImage)
+	buildCtx, err := createBuildContext(dockerfile)
+	assert.NilError(t, err)
+	defer os.RemoveAll(buildCtx)
+
+	validCases := []struct {
+		name     string
+		network  string
+		exitCode int
+	}{
+		{"test with empty network", "", 0},
+		{"test with default network", "default", 0},
+		// When network=none, can't connect to internet, therefore cannot download packages in the dockerfile
+		{"test with no network", "none", 1},
+	}
+
+	for _, tc := range validCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			base.Cmd("build", buildCtx, "--network", tc.network).AssertExitCode(tc.exitCode)
```

Comment:
Please try adding `--no-cache` to fix the CI failure

Reply:
And it is weird that the test is passing for `nerdctl` without `--no-cache`. 🤔

Reply:
> Please try adding --no-cache to fix the CI failure with docker.
Or, test --network=none before running other tests

This did the trick although not sure why it works with `nerdctl` without this option. Also works with docker on lima instance on my local. I'll add a comment about it.